src/Command/CodeCheckerCommand.php

```diff
@@ -63,106 +45,69 @@ protected function configure()
     protected function initialize(InputInterface $input, OutputInterface $output)
     {
         parent::initialize($input, $output);
-
-        // Resolve the coding standard.
-        $resolver       = new StandardResolver();
-        $this->standard = $input->getOption('standard');
-        if ($resolver->hasStandard($this->standard)) {
-            $this->standard = $resolver->resolve($this->standard);
-        }
-
-        $this->finder = Finder::create()->name('*.php');
+        $this->initializeExecute($output, $this->getHelper('process'));
+        $this->summaryFile = sys_get_temp_dir().'/moodle-plugin-ci-code-checker-summary-'.time();
     }
 
     protected function execute(InputInterface $input, OutputInterface $output)
     {
-        $this->outputHeading($output, 'Moodle Code Checker on %s');
+        $this->outputHeading($output, 'Moodle CodeSniffer standard on %s');
 
-        $files = $this->plugin->getFiles($this->finder);
+        $files = $this->plugin->getFiles(Finder::create()->name('*.php'));
         if (count($files) === 0) {
             return $this->outputSkip($output);
         }
 
-        // Needed constant.
-        if (defined('PHP_CODESNIFFER_CBF') === false) {
-            define('PHP_CODESNIFFER_CBF', false);
+        $builder = ProcessBuilder::create()
+            ->setPrefix('php')
+            ->add(__DIR__.'/../../vendor/squizlabs/php_codesniffer/bin/phpcs')
+            ->add('--standard='.($input->getOption('standard') ?: 'moodle'))
+            ->add('--extensions=php')
+            ->add('-p')
+            ->add('-w')
+            ->add('-s')
+            ->add('--no-cache')
+            ->add($output->isDecorated() ? '--colors' : '--no-colors')
+            ->add('--report-full')
+            ->add('--report-width=132')
+            ->add('--encoding=utf-8')
+            ->setWorkingDirectory($this->plugin->directory)
+            ->setTimeout(null);
+
+        // If we aren't using the max-warnings option, then we can forget about warnings and tell phpcs
+        // to ignore them for exit-code purposes (still they will be reported in the output).
+        if ($input->getOption('max-warnings') < 0) {
+            $builder->add('--runtime-set')->add('ignore_warnings_on_exit')->add(' 1');
+        } else {
+            // If we are using the max-warnings option, we need the summary report somewhere to get
+            // the total number of errors and warnings from there.
+            $builder->add('--report-summary='.$this->summaryFile);
```

Comment:
Thinking if the better approach could be using `--report=json`, from which the number of warnings and errors can be retrieved without need of string parsing?

Reply:
Done, I've added a 3rd, fixup commit, with the switch from summary to json report format to get the total errors and warnings.